packages/iframe-channel-provider/src/postmessage-service.ts

```diff
@@ -58,21 +63,24 @@ export class PostMessageService {
   private requestNumber = 0;
   async request<ResultType = any>(
     target: Window,
-    message: JsonRpcRequest,
+    messageWithOptionalId: Omit<JsonRpcRequest, 'id'> & {id?: number}, // Make id an optional key
     callback?: (result: ResultType) => void
   ): Promise<ResultType> {
     // Some tests rely on being able to supply the id on the message
     // We should not allow this in production, as we cannot guarantee unique
     // message ids.
-    if (message.id) logger.error('message id should not be defined');
+    if (messageWithOptionalId.id) logger.error('message id should not be defined');
 
     // message IDs should be unique
-    message.id = message.id || this.requestNumber++;
+    const message: JsonRpcRequest = {
+      ...messageWithOptionalId,
+      id: messageWithOptionalId.id || this.requestNumber++
+    };
```

Comment:
This approach allows the `JsonRpcRequest` type to insist on the `id` field being specified.